packages/zwave-js/src/lib/controller/Controller.ts

```diff
@@ -1885,6 +1885,21 @@ ${associatedNodes.join(", ")}`,
 	 * @param nodeId The id of the node to remove
 	 */
 	public async removeFailedNode(nodeId: number): Promise<void> {
+		const node = this.nodes.get(nodeId);
+		if (!node) {
+			throw new ZWaveError(
+				`Node ${nodeId} was not found!`,
+				ZWaveErrorCodes.Controller_NodeNotFound,
+			);
+		}
+
+		if (await node.ping()) {
+			throw new ZWaveError(
+				`The node removal process could not be started because the node responded to a ping.`,
+				ZWaveErrorCodes.ReplaceFailedNode_Failed,
+			);
+		}
```

Comment:
This is repeated later, couldn't this be a new private function like `pingNode` taht returns a node?

Reply:
I don't follow?